components/admin_console/team_channel_settings/channel/details/channel_members/index.ts

```diff
@@ -51,7 +51,7 @@ type Actions = {
 };
 
 function searchUsersToAdd(users: Dictionary<UserProfile>, term: string): Dictionary<UserProfile> {
-    const profiles = filterProfilesMatchingTerm(Object.keys(users).map((key) => users[key]), term);
+    const profiles = filterProfilesStartingWithTerm(Object.keys(users).map((key) => users[key]), term);
```

Comment:
Could this also be replaced with
```suggestion
    const profiles = filterProfilesStartingWithTerm(Object.values(users), term);
```